### ISSUE TITLE
Dynamic Resource Generator Testing | Empty resource generator code region in MSFT_[resource].psm1

### ISSUE THREAD

**Issue by TheBrickAdmin** [microsoft/Microsoft365DSC#3211]
#### Details of the scenario you tried and the problem that is occurring
I have started working with the Dynamic Resource Generator by creating my own PowerShell script to use this module.

So far I have been able to work around a lot of errors I initially got where files weren't on their expected locations and am able to generate the required resources. Within these resources there are errors in the MSFT_[resource].psm1 files where the resource generator code region in very function is empty except for a "]" and a "$,".

Most likely this is due to my lack of understanding, but I cannot seem to find what I am doing wrong here.

#### Verbose logs showing the problem
Example of the first lines of any function:
```
function Get-TargetResource
{
    [CmdletBinding()]
    [OutputType([System.Collections.Hashtable])]
    param
    (
        #region resource generator code
        
]
        $,

        #endregion

        [Parameter()]
        [System.String]
        [ValidateSet('Absent', 'Present')]
        $Ensure = 'Present',
```

#### Suggested solution to the issue
N/A

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
I have attached the script I have used thus far.
[Microsoft365DSCResourceGenerator.txt](https://github.com/microsoft/Microsoft365DSC/files/11320172/Microsoft365DSCResourceGenerator.txt)

#### The operating system the target node is running
OsName               : Microsoft Windows 11 Enterprise
OsOperatingSystemSKU : EnterpriseEdition
OsArchitecture       : 64-bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 22621.1.amd64fre.ni_release.220506-1250
OsLanguage           : en-US
OsMuiLanguages       : {en-US, en-GB}

#### Version of the DSC module that was used ('dev' if using current dev branch)
dev

**Comment by ykuijs** [comment]
@William-Francillette, could you please help out @TheBrickAdmin with this question? Since you have a lot of experience with the DRG :wink:

**Comment by andikrueger** [comment]
Actually, I just ran a test with the latest DRG Version and was able to properly create the psm1 files.

![image](https://user-images.githubusercontent.com/15608729/234219028-80080df7-9e50-42f9-aec2-8520892f2f7d.png)


Please do not open the file on the first creation. You need to wait a bit until the DRG is done with the magic. ;)

**Comment by TheBrickAdmin** [comment]
I have removed the old configuration and tried again (this involves cloning the repo, so I am using the latest code).

During execution I get the following errors:
_InvalidOperation: You cannot call a method on a null-valued expression.
InvalidOperation: You cannot call a method on a null-valued expression.                                                 
Get-M365DSCFakeValues: Cannot bind argument to parameter 'ParametersInformation' because it is null.                    
Get-M365DSCFakeValues: Cannot bind argument to parameter 'ParametersInformation' because it is null.                    
Get-M365DSCHashAsString: Cannot bind argument to parameter 'Values' because it is null.                                 
Get-M365DSCHashAsString: Cannot bind argument to parameter 'Values' because it is null.                                 
Get-M365DSCHashAsString: Cannot bind argument to parameter 'Values' because it is null.
Get-M365DSCFakeValues: Cannot bind argument to parameter 'ParametersInformation' because it is null.                    
Get-M365DSCHashAsString: Cannot bind argument to parameter 'Values' because it is null.
Get-M365DSCDRGParameterTypeForSchema: Cannot bind argument to parameter 'Type' because it is an empty string._

The output is not like yours @andikrueger after several minutes of waiting.
![image](https://user-images.githubusercontent.com/73339591/234224222-5e5ea5c8-4727-4595-9cd7-3d8e23dfdea4.png)

Most likely I am doing something wrong. I am just not sure what.

**Comment by andikrueger** [comment]
Please run Uninstall-M365DSCOutdatedDependencies. It is possible, there is a mixup of various versions of the MS Graph Modules.

I also get some of the errors above. This has to do with some advanced resolving of the additional parameters.

**Comment by William-Francillette** [comment]
@TheBrickAdmin Can you share you New-M365DSCResource command please?

**Comment by William-Francillette** [comment]
Ok just found it
```
New-M365DSCResource -ResourceName AADUser -Workload MicrosoftGraph -CmdletNoun 'MgDomain' -Path $resourceDestinationPath -ExampleFilePath $exampleSourcePath -Credential $creds
New-M365DSCResource -ResourceName AADDomain -Workload MicrosoftGraph -CmdLetNoun 'MgDomain' -Path $resourceDestinationPath -ExampleFilePath $exampleSourcePath -Credential $creds
```

The DRG is still in preview and may not work for all type of resources - I mainly focused on generating Intune resources - I've also used it for AAD but didn't pushed it too much to be honest

At present, the DRG will allow you to have a baseline for your resource but still requires debugging and adaption for some resources
I haven't tested the MgDomain generation so not sure why the test unit (Get-M365DSCFakeValues) is not working and why it didn't retrieve the parameters

**Comment by TheBrickAdmin** [comment]
@William-Francillette Could you share one of the New-M365DSCResource commands you used? This would make sure I test the same thing as you did, so we can rule out a mistake in calling the command at my end.

**Comment by andikrueger** [comment]
I used this command:

```powershell
$ResourcePath = "C:\DEV\Microsoft365DSC\Modules\Microsoft365DSC\DSCResources"
$UnitTestPath = "C:\DEV\Microsoft365DSC\Tests\Unit\Microsoft365DSC"
$ExamplePath = "C:\DEV\Microsoft365DSC\Modules\Microsoft365DSC\Examples\Resources"
 
New-M365DSCResource -ResourceName AADDomain -Workload MicrosoftGraph -CmdLetNoun 'MgDomain' -Path $ResourcePath
```

**Comment by TheBrickAdmin** [comment]
Thx @andikrueger. I have tested with your exact commands (just used my Assert-Path function to make sure the paths existed and copied the required templates). Yet I still get the same error/output.
![image](https://user-images.githubusercontent.com/73339591/234237877-2f4ff40a-ed1a-4a02-b813-14e7d41ff5c8.png)

**Comment by William-Francillette** [comment]
```
$ResourceName="IntuneDeviceConfigurationEditionUpgradePolicyWindows10"
$ResourcePath = "C:\DevOps\Microsoft365DSC-1\Modules\Microsoft365DSC\DSCResources"
$UnitTestPath = "C:\DevOps\Microsoft365DSC-1\Tests\Unit\Microsoft365DSC"
$ExamplePath = "C:\DevOps\Microsoft365DSC-1\Modules\Microsoft365DSC\Examples\Resources"
cd C:\DevOps\Microsoft365DSC-1\ResourceGenerator
Get-Module|remove-Module -Force
Import-Module C:\DevOps\Microsoft365DSC-1\ResourceGenerator\M365DSCResourceGenerator.psm1 -force

New-M365DSCResource -ResourceName "$ResourceName" `
                -Workload MicrosoftGraph `
                -CmdletNoun 'MgDeviceManagementDeviceConfiguration' `
                -Path $ResourcePath `
                -UnitTestPath $UnitTestPath `
                -ExampleFilePath $ExamplePath `
                -APIVersion beta `
                -Credential $credential `
                -AdditionalPropertiesType 'editionUpgradeConfiguration' `
                -ParametersToSkip @(
                    #'accessPackage'
                    #'accessPackageCatalog' 
                    #"AccessPackageResourceRoles"
                    #'AccessPackageResourceScopes'
                    #'AccessPackageResources'
                    #'AccessPackages'
                    'Version'
                    'SupportsScopeTags'
                    'RoleScopeTagIds'
                    'deviceManagementApplicabilityRuleDeviceMode'
                    'DeviceManagementApplicabilityRuleOsEdition'
                    'DeviceManagementApplicabilityRuleOsVersion'
                    'ModifiedBy'
                    'LastModifiedDateTime'
                    'ModifiedDateTime'
                    'CreatedBy'
                    'createdDateTime'
                    'endOfSupportDate'
                    'DeployableContentDisplayName'
                    )
```

This what I normally use

**Comment by TheBrickAdmin** [comment]
And I think I actually found my issue... I was using PowerShell Core 7.3.4. When I use Windows Powershell 5.1 The command executes as expected.

**Comment by William-Francillette** [comment]
Fair enough - the support for PS 7 is not fully implemented yet unfortunately - I should have mentioned to use 5.1

**Comment by TheBrickAdmin** [comment]
Thx for bearing with me during this ordeal. I will close this issue since you mention support for PS7 isn't fully implemented yet, so this isn't an issue currently in scope of the project.

Glad you were willing to help.

**Comment by andikrueger** [comment]
Thanks for the feedback. PS5.1 is the way to go with DSC.
